### PR TITLE
Perf: fuse and right-size the DSpark decode-forward head tail

### PR DESCRIPTION
**Pull request by zhangqi-chen**
- Size the LM head at one owner's rows. MAX_LOGIT_ROWS was the whole
  step's DECODE_TOKENS, but a TP group splits the step across its ranks
  and a rank only ever owns MOE_TOKENS of them, so the projection ran
  four times the real rows and the logits window was four times the
  size it needs. Key it off MOE_TOKENS, which the forward entries do
  not rewrite at import the way they rewrite config.TP.
- Fuse the projection and the push back into one mixed cube+vector
  scope: the cube projects a vocab tile, pl.aiv_shard carries the
  accumulator across the C->V edge, and pld.tensor.remote_store pushes
  each owner's rows to that peer's window while the cube projects the
  next tile. This replaces the GM-staged lm_head_matmul plus put-based
  lm_head_combine_push, and returns the logits window to two
  dimensions so the combine gather copies row blocks instead of one
  pl.load/pl.store per row.
- Widen the LM head tiles: one whole owner per matmul row block is the
  widest M the Acc space admits and halves the weight tile re-reads,
  and the hidden gather moves 1024-wide blocks of 64 rows.
- Fold the inactive-row mask into greedy_sample, so a row without a
  live logit index costs no vocabulary scan instead of being
  overwritten by a second pass over sampled_ids. Drops
  mask_inactive_sample_rows from both forward entries.
- Add an l2/l3 lm_head sampling entry behind --entry sample, so the
  projection-plus-sampling tail has a standalone fixture and
  greedy_sample has a golden.
- Retile hc_head for the decode-forward tail. The vector array is 48
  cores wide, so give the rms stage its own taller row tile and widen
  the reduce D slice to put both fan-outs at 48 tasks rather than four
  waves of short ones, and stop splitting the cube stage over K, where
  the row blocks already fill the array.
- Expose --enable-chip-swimlane on the dspark decode forward.

LM head, tp4 with 128 active rows on a2a3, 20 rounds: 9030.8 -> 1470.3
us headline mean, 8264.8 -> 1016.9 us on the fastest rank. hc_head at
384 rows, 50 rounds: 439.4 -> 83.7 us mean.

### CODE REVIEW

File: models/deepseek_v4_flash_dspark/lm_head.py [hunk 1]
```diff
@@ -86,6 +92,7 @@ def _parse_int_argv(name, default=None):
 assert MAX_LOGIT_ROWS % MM_ROW_TILE == 0, "each row block must be one owner's rows"
 assert MAX_LOGIT_ROWS % PUSH_ROW_TILE == 0, "row block must cover whole rows"
 assert MAX_LOGIT_ROWS % LOGITS_GATHER_ROW_TILE == 0, "logits row blocks must cover whole rows"
+assert TP_SIZE % AIV_LANES == 0, "owners must divide evenly across the AIV lanes"
```
[the remark this answers is not in the document]
Comment (zhangqi-chen) [reply models/deepseek_v4_flash_dspark/lm_head.py]: Fixed in `1fe407b2` — relaxed to `TP_SIZE == 1 or TP_SIZE % AIV_LANES == 0`. The invariant the assertion guards is that the peer-notify loop pairs owners across the two lanes; at TP1 that loop is `pl.range(0)` and there is no peer to notify, so the pairing requirement does not apply. Verified `--tp 1 --dp 1` passes on a2a3 for both the projection and `--entry sample`.
